uaclient/files/state_files.py

```diff
@@ -0,0 +1,50 @@
+from uaclient.data_types import DataObject, Field, StringDataValue, data_list
+from uaclient.exceptions import InvalidStateFile
+from uaclient.files.data_types import DataObjectFile
+from uaclient.files.files import UAFile
+
+SERVICES_ONCE_ENABLED = "services-once-enabled"
+
+
+class ServicesOnceEnabledData(DataObject):
+    fields = [
+        Field("services_once_enabled", data_list(StringDataValue), False)
+    ]
+
+    def __init__(self, services_once_enabled):
+        self.services_once_enabled = services_once_enabled
+
+
+class ServiceOnceEnableDataObjectFile(DataObjectFile):
+    def update_services(self, service: str):
+        file_content = self.read() or {}
+        self.write(
+            ServicesOnceEnabledData(
+                services_once_enabled=list(
+                    set(
+                        file_content.get("services_once_enabled", [])
+                        + [service]
+                    )
+                )
+            )
+        )
```

Comment:
I'm not sure if the specialized class/method is necessary.

Without the specialized class/method, use would look like this (using the field name "service_names" and assuming it is required as suggested above):
```python
services_once_enabled_file = DataObjectFile(
    data_object_cls=ServicesOnceEnabledData,
    ua_file=UAFile(
        name=SERVICES_ONCE_ENABLED,
        private=False,
    ),
)
# all of this is checked by mypy
services_once_enabled_do = services_once_enabled_file.read() or ServicesOnceEnabledData(service_names=[])
services_once_enabled_do.service_names = list(set(services_once_enabled_do.service_names + ["fips-updates"]))
services_once_enabled_file.write(services_once_enabled_do)
```

With the specialized class/method, it needs to be adjusted to specify the generic type of DataObjectFile so that mypy understands what is going on and can help us. Something like this:
```suggestion
class ServiceOnceEnableDataObjectFile(DataObjectFile[ServicesOnceEnabledData]):
    def add_service(self, service: str):
        file_content = self.read() or ServicesOnceEnabledData(service_names=[])
        self.write(
            ServicesOnceEnabledData(
                service_names=list(
                    set(
                        file_content.service_names
                        + [service]
                    )
                )
            )
        )
```